### PR TITLE
Selector Parser Options

### PR DESCRIPTION
**Pull request by wcharczuk**
## PR Summary

Adds many more knobs to turn with selector parsing.

 - **Type:** Features
 - **Intended Change Level:** minor

#### Reviewers:

Reviewers keep the following in mind while reviewing this PR, and provide an assessment of them in your approval comment:

- Does the "Intended Change Level" above match the actual behavior of this PR?
- Is this PR following patterns set forth in the package (if modifying a package), or the go-sdk design patterns if implementing a new package from scratch?
- Does this require a backport to LTS versions? If so ping, let the contributors group know.

### CODE REVIEW

File: ex/exception.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func (e *Ex) Format(s fmt.State, verb rune) {
 		if len(e.Message) > 0 {
 			io.WriteString(s, "; "+e.Message)
 		}
-		if s.Flag('+') {
+		if s.Flag('+') && e.StackTrace != nil {
```
Comment (dhermes) [inline ex/exception.go]: GitHub why you put this in the diff? It's already in `master`: https://github.com/blend/go-sdk/blob/6051c0b95d8d63733ac88802789a93a72b87ec07/ex/exception.go#L113

@wcharczuk Branch may need a rebase?

File: selector/check.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		if err = CheckValue(value); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// CheckKey validates a key with the default validation options.
+func CheckKey(key string) error {
+	return new(Parser).CheckKey(key)
```
Comment (dhermes) [inline selector/check.go]: Any reason to prefer `new(Parser)` over `&Parser{}`? (Personally I avoid `new()`)

File: selector/check_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
+package selector
+
+import (
+	"strings"
+	"testing"
+
+	assert "github.com/blend/go-sdk/assert"
```
Comment (dhermes) [inline selector/check_test.go]: Why the no-op import alias?

File: selector/selector.go [hunk 4]
```diff
@@ -6,6 +6,5 @@ type Labels = map[string]string
 // Selector is the common interface for selector types.
 type Selector interface {
 	Matches(labels Labels) bool
-	Validate() error
```
Comment (dhermes) [inline selector/selector.go]: Why'd you nuke `Validate()`?

File: ex/exception.go (hunk 1 above)
Comment (wcharczuk) [reply ex/exception.go]: this branch is super old

File: selector/check.go (hunk 2 above)
Comment (wcharczuk) [reply selector/check.go]: `new()` is clearer for me (specifically around if it will heap allocate)

File: selector/check_test.go (hunk 3 above)
Comment (wcharczuk) [reply selector/check_test.go]: `goimports` is annoying

File: selector/selector.go (hunk 4 above)
Comment (wcharczuk) [reply selector/selector.go]: It's done when parsing, not on the final AST node

**Pull request closed without merging**